src/main/java/com/fasterxml/jackson/databind/cfg/DefaultCacheProvider.java

```diff
@@ -163,13 +189,32 @@ public Builder maxSerializerCacheSize(int maxSerializerCacheSize) {
             return this;
         }
 
+        /**
+         * Define the maximum size of the {@link LookupCache} instance constructed by {@link #forTypeFactory()}
+         * and {@link #_buildCache(int)}
+         * <p>
+         * Note that value zero is not considered a magic number, unlike some other libraries where it means "disabling cache".
+         * Instead setting value to zero only means setting cache size to zero.
```

Comment:
I would rephrase... For DefaultCacheProvider, zero max size effectively does disable caching since entries are not retained in the underlying LRUMap. Yes, it's technically true that a write & removal occurs under the hood, but users don't care about this detail (other than for optimizing performance, as a special no-op implementation could've been used instead)

Perhaps: `Specifying a maximum size of zero prevents reuse of values that would have been cached`